src/detectors/builtin/stateMutationInGetter.ts

```diff
@@ -0,0 +1,201 @@
+    }
+
+    return warnings;
+  }
+
+  /**
+   * Checks if a getter function contains state-mutating function calls or direct mutations
+   */
+  private checkGetterForStateMutations(
+    cu: CompilationUnit,
+    getter: AstFunctionDef,
+    contractName: string,
+    warnings: MistiTactWarning[],
+  ): void {
+    // Direct state mutations
+    this.checkDirectStateMutations(getter, warnings);
+
+    // Indirect state mutations through function calls
+    forEachExpression(getter, (expr: AstExpression) => {
+      if (expr.kind === "static_call") {
+        this.checkStaticCall(cu, expr, contractName, warnings);
+      } else if (expr.kind === "method_call") {
+        this.checkMethodCall(cu, expr, contractName, warnings);
+      }
+    });
+  }
+
+  /**
+   * Direct state mutations in statements
+   */
+  private checkDirectStateMutations(
+    getter: AstFunctionDef,
+    warnings: MistiTactWarning[],
+  ): void {
+    foldStatements(
+      getter,
+      (acc, stmt) => {
+        const mutations = collectMutations(stmt);
+        if (mutations && mutations.mutatedFields.length > 0) {
+          acc.push(
+            this.makeWarning(
+              "Getter directly mutates contract state",
+              stmt.loc,
+              {
+                extraDescription:
+                  "Getter methods should not modify state directly or indirectly",
+                suggestion:
+                  "Move state-modifying logic to a non-getter function",
+              },
+            ),
+          );
+        }
+        return acc;
+      },
+      warnings,
+    );
+  }
+
+  /**
+   * Checks if a static function call might modify state
+   */
+  private checkStaticCall(
+    cu: CompilationUnit,
+    call: AstStaticCall,
+    contractName: string,
+    warnings: MistiTactWarning[],
+  ): void {
+    const calleeName = idText(call.function);
+    const calleeNodeId = cu.callGraph.getNodeIdByName(calleeName);
+
+    if (calleeNodeId !== undefined) {
+      const calleeNode = cu.callGraph.getNode(calleeNodeId);
+      if (calleeNode && calleeNode.hasEffect(Effect.StateWrite)) {
+        warnings.push(
+          this.makeWarning(
+            `Getter calls state-mutating function: ${calleeName}`,
+            call.loc,
+            {
+              extraDescription:
+                "Getter methods should not modify state directly or indirectly",
+              suggestion: "Move state-modifying logic to a non-getter function",
+            },
+          ),
+        );
+      }
+    }
+  }
+
+  /**
+   * Checks if a method call might modify state
+   */
+  private checkMethodCall(
+    cu: CompilationUnit,
+    call: AstMethodCall,
+    contractName: string,
+    warnings: MistiTactWarning[],
+  ): void {
+    // For self.method() calls check the contract's method
+    if (call.self.kind === "id" && idText(call.self) === "self") {
+      const methodName = idText(call.method);
+      const fullMethodName = `${contractName}::${methodName}`;
```

Comment:
This is unreliable since the naming logic will be changed soon. Replace w/ a function from `callgraph.ts`.